papas/pfalgo/pfblock.py

```diff
@@ -36,11 +36,9 @@ def __init__(self, element_ids, edges, subtype):
            
         '''
         #make a uniqueid for this block
-        self.uniqueid = Identifier.make_id(Identifier.PFOBJECTTYPE.BLOCK, subtype) 
-        self.element_uniqueids = sorted(element_ids, key=lambda  x: Identifier.type_letter(x)) 
-
-        #comment out energy sorting  for now as not available C++
-        sortby = lambda x: Identifier.type_letter(x)
+        self.uniqueid = Identifier.make_id(Identifier.PFOBJECTTYPE.BLOCK, index, subtype, len(element_ids))
+        #this will sort by type eg ecal, hcal, track and then by energy (biggest first)
+        sortby = lambda x: (Identifier.type_letter(x), -x)
```

Comment:
Can we discuss this... perhaps use  Identifier.get_value(x) instead?  also see below.